engine/script/src/luacjson/lua_cjson.c

```diff
@@ -532,8 +533,9 @@ static void json_initialize_config(json_config_t* cfg)
 static void json_encode_exception(lua_State *l, json_config_t *cfg, strbuf_t *json, int lindex,
                                   const char *reason)
 {
-    if (!cfg->encode_keep_buffer)
+    if (cfg->encode_keep_buffer)
         strbuf_free(json);
```

Comment:
We trow an error here using `luaL_error` which will long_jmp and release the stack
We expect memory will be freed if error thrown:
https://github.com/defold/defold/blob/9123aa9aa1aa26953d396ddc8256b8d518fe805e/engine/script/src/script_json.cpp#L154-L160